### PR TITLE
Add ComputePlugins class

### PR DESCRIPTION
**Pull request by bloriot97**
In order to get some useful information about checkstyle (parameters, rule-set...) I need to compute maven's plugins. 

I thought it might be useful to implement it as a step in case some else wants to re-use it. 

I have also added the test cases.

### CODE REVIEW

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/paths/ComputePlugins.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import fr.inria.spirals.repairnator.process.step.AbstractStep;
+import fr.inria.spirals.repairnator.process.step.StepStatus;
+import org.apache.maven.model.Model;
+import org.apache.maven.model.Plugin;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ComputePlugins extends AbstractStep {
```
Comment (monperrus) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/paths/ComputePlugins.java]: javadoc?

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/inspectors/properties/projectMetrics/ProjectMetrics.java [hunk 2]
```diff
@@ -7,6 +7,11 @@
      * {@link fr.inria.spirals.repairnator.process.step.paths.ComputeModules}.
      */
     private int numberModules;
+    /**
+     * {@code numberPluging} is calculated in the step
+     * {@link fr.inria.spirals.repairnator.process.step.paths.ComputePlugins}.
+     */
+    private int numberPluging;
```
Comment (fermadeiral) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/inspectors/properties/projectMetrics/ProjectMetrics.java]: Fix name of variable to `numberPlugins`.

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/paths/ComputePlugins.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    private List<Plugin> findPlugins(String pomPath) {
+        List<File> plugins = new ArrayList<>();
+
+        File pomFile = new File(pomPath);
+        Model model = MavenHelper.readPomXml(pomFile, this.getInspector().getM2LocalPath());
+        if (model == null) {
+            this.addStepError("Error while building model: no model has been retrieved.");
+            return null;
+        }
+        if (model.getBuild().getPlugins() == null) {
```
Comment (fermadeiral) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/paths/ComputePlugins.java]: `model.getBuild()` can be null.

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/paths/ComputePlugins.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    @Override
+    protected StepStatus businessExecute() {
+        this.getLogger().debug("Computing project plugins...");
+
+        String mainPomPath = this.getPom();
+        List<Plugin> plugins = this.findPlugins(mainPomPath);
+
+        if (plugins == null || plugins.size() == 0) {
+            this.getLogger().info("No plugins was found.");
+            // return StepStatus.buildError(this, PipelineState.PLUGINSNOTCOMPUTED);
```
Comment (fermadeiral) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/paths/ComputePlugins.java]: Why is this line commented?

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/paths/ComputePlugins.java (hunk 1 above)
Comment (bloriot97) [reply repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/paths/ComputePlugins.java]: Is it better now ?

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/inspectors/properties/projectMetrics/ProjectMetrics.java [hunk 5]
```diff
@@ -7,6 +7,11 @@
      * {@link fr.inria.spirals.repairnator.process.step.paths.ComputeModules}.
      */
     private int numberModules;
+    /**
+     * {@code numberPluging} is calculated in the step
+     * {@link fr.inria.spirals.repairnator.process.step.paths.ComputePlugins}.
+     */
+    private int numberPlugings;
```
Comment (fermadeiral) [inline repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/inspectors/properties/projectMetrics/ProjectMetrics.java]: There is an extra "g".

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/paths/ComputePlugins.java (hunk 3 above)
Comment (bloriot97) [reply repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/paths/ComputePlugins.java]: I have added a check before.

File: repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/inspectors/properties/projectMetrics/ProjectMetrics.java (hunk 5 above)
Comment (bloriot97) [reply repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/inspectors/properties/projectMetrics/ProjectMetrics.java]: Oh yeah sorry my bad.

**Comment by fermadeiral** [comment]
This PR looks good to me.

**Comment by bloriot97** [comment]
Perfect, @btellstrom can you merge it ?

**Comment by btellstrom** [comment]
Looks good to me as well.

**Pull request merged**